### ISSUE TITLE
False negative E262 inline comment should start with '# ' when there is a non breaking whitespace after a space

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pycodestyle#1034]
It happens often that you start typing a ``#`` then keep typing on the key longer than required and create a non breaking whitespace instead of a space. Then old version of ``black`` added a space at the beginning and there would be one space and one non-breaking white space (NBSP) at the start of the comment.

```python
a = 42 #  (One space one NBSP)
```
Should raise a E262 like
```python
a = 42 #  (Two spaces)
```
does.

I'm willing to contribute the evolution if this is a desirable change.

**Comment by sigmavirus24** [comment]
I believe GitHub has converted what you're trying to demonstrate to ascii. Could you upload a file instead with the two examples?

**Comment by Pierre-Sassoulas** [comment]
Right, sorry about that. Fun fact, github does not handle uploading of python files and pastebin also convert nbsp to normal spaces, which is why I upload a ``.txt`` and not a ``.py``

[nbsp.txt](https://github.com/PyCQA/pycodestyle/files/7474591/nbsp.txt)

If this first line still contains normal spaces, could you please create the space yourself following instructions [here](https://www.computerhope.com/jargon/n/nbsp.htm) ?

**Comment by sigmavirus24** [comment]
That worked fine:

```
❯ hexdump -C nbsp.py 
00000000  61 20 3d 20 34 32 20 23  20 c2 a0 28 4f 6e 65 20  |a = 42 # ..(One |
00000010  73 70 61 63 65 20 6f 6e  65 20 4e 42 53 50 29 0d  |space one NBSP).|
00000020  0a 61 20 3d 20 34 32 20  23 20 20 28 54 77 6f 20  |.a = 42 #  (Two |
00000030  73 70 61 63 65 73 29                              |spaces)|
00000037
```

Makes sense to fix this in my opinion. A file like this would be a good case to use to test with

**Comment by Pierre-Sassoulas** [comment]
Neat, I'm going to work on it then ! :)

**Comment by Pierre-Sassoulas** [comment]
Small question, it seems that ``#  (Two spaces) `` is permissible for block comment so should we warn for block comment starting with an NBSP ?

```python
#: (E265:1:1?) E262:2:9
#  (One space one NBSP) Should be ok as two spaces is ok?
a = 42  #  (One space one NBSP)
#: E262:2:9
#  (Two spaces) Ok for block comment 
a = 42  #  (Two spaces)
```

**Comment by asottile** [comment]
from the comments, block comments should follow the same 1-space rule

**Comment by Pierre-Sassoulas** [comment]
I can do that but it means block comments like this would start raising E265:
```python
            # Subtleties:
            # - we don't *completely* ignore the last line; if it
            #   contains the magical "# noqa" comment, we disable all # E265
            #   physical checks for the entire multiline string # E265
            # - have to wind self.line_number back because initially it
            #   points to the last line of the string, and we want # E265
            #   check_physical() to give accurate feedback # E265
```

**Comment by asottile** [comment]
then the comment should be corrected 🤷

**Comment by Pierre-Sassoulas** [comment]
Well yes, but I means it this comment really wrong ? A list is easier to read with a small indentation for each element.

**Comment by asottile** [comment]
I mean the source code comment which says that it should start with exactly 1 space
